components/exceptions.py

```diff
@@ -0,0 +1,38 @@
+import typing
+from fastapi import status as http_status
+from components.enums import JSENDStatus
+
+
+class BackendException(Exception):
+    def __init__(
+        self,
+        *,
+        status: JSENDStatus = JSENDStatus.FAIL,
+        data: typing.Union[None, int, str, list, dict] = None,
+        message: str,
+        code: int = http_status.HTTP_400_BAD_REQUEST
+    ):
+        self.status = status
+        self.data = data
+        self.message = message
+        self.code = code
```

Comment:
It's a good feature of course. But how can it help us? How many string will we reduce? And how will change the readability of code? But if you insist I can do it.

Reply:
I have not counted amount of lines, but it usually remove necessity to implement plenty of methods (`__init__`, `__repr__`, `__str__` and `.dict()` in this case). It's the main purpose of dataclasses

> And how will it change the readability of code?

It may remove methods in class definition (at first glance — all of them) and may require to replace `.dict()` calls with `.asdict()` (unless one define an alias `.dict()` → `.asdict()`)

I do not insist, just pointing out.

BTW, @Kostiantyn-Salnykov could pydantic be used for similar purpose?